.github/workflows/package-binaries.yml

```diff
@@ -138,58 +138,28 @@ jobs:
-          draft: false
-          prerelease: true
-
-      - uses: actions/download-artifact@v1
+    steps:
+      - uses: actions/download-artifact@v3
         with:
           name: spinalcordtoolbox-binaries_linux
-          
-      - uses: actions/download-artifact@v1
+
+      - uses: actions/download-artifact@v3
         with:
           name: spinalcordtoolbox-binaries_osx
 
-      - uses: actions/download-artifact@v1
+      - uses: actions/download-artifact@v3
         with:
           name: spinalcordtoolbox-binaries_windows
 
-      - name: Upload Release Asset
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps 
-          asset_path: ./spinalcordtoolbox-binaries_linux/spinalcordtoolbox-binaries_linux.tar.gz
-          asset_name: spinalcordtoolbox-binaries_linux.tar.gz
-          asset_content_type: application/gzip
-          
-      - name: Upload Release Asset
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps 
-          asset_path: ./spinalcordtoolbox-binaries_osx/spinalcordtoolbox-binaries_osx.tar.gz
-          asset_name: spinalcordtoolbox-binaries_osx.tar.gz
-          asset_content_type: application/gzip
-
-      - name: Upload Release Asset
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Create Release
+        uses: ncipollo/release-action@v1
         with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
-          asset_path: ./spinalcordtoolbox-binaries_windows/spinalcordtoolbox-binaries_windows.tar.gz
-          asset_name: spinalcordtoolbox-binaries_windows.tar.gz
-          asset_content_type: application/gzip
+          token: ${{ secrets.GITHUB_TOKEN }}
+          # name the release with the run_id to allow multiple builds on the same branch/tag
+          # https://github.com/actions/create-release/issues/2#issuecomment-613591846
+          tag: ${{ github.ref }}-${{github.run_id }}
```

Comment:
Oh yeah, looking at [the list of tags](https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/tags), it seems that `actions/create-release` allowed tag names that start with `refs/`, even though `ncipollo/release-action` doesn't allow this.

According to [the list of variables available](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context), maybe `github.ref_name` is good enough? Because `github.ref` will always start with the illegal string `refs/`.
```suggestion
          tag: ${{ github.ref_name }}-${{github.run_id }}
```

Reply:
Alternatively, should we even be running the `create-release` action on PRs?

(Compared to a number of other release workflows we maintain (e.g. SCT, sct_tutorial_data, build-ants, etc.), we tend to do everything _but_ the actual release creation in those other workflows, presumably for this exact reason.)

In fact, all those `refs/` tags in this repo feel like a bit of a smell to me? I'm wondering if they should even have been created at all, given all the non-essential clutter.

I guess the real question I have is: What is the "create-release" step even testing on PRs? (That the assets were successfully created? Well, that's kind of already covered by the "upload artifact" step, no?)

Reply:
And, if we skip running the create-release step on PRs, then I think that might help to  simplify some of the logic you've added (namely: avoiding duplicate releases on the same commit).

Reply:
Sure! Feel free to push more commits if you'd rather change the flow of the workflow.

I suspect that actually creating a release on each PR was useful to manually inspect the created artifacts, early on in the life of this repository. But at this point it just seems like clutter.

Reply:
> I suspect that actually creating a release on each PR was useful to manually inspect the created artifacts, early on in the life of this repository. But at this point it just seems like clutter.

Exactly. And, if we do still want to inspect the artifacts, they're still accessible from the [Summary page](https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/actions/runs/6803920799?pr=6), even if we never run the create-release step:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/assets/16181459/89a305ce-62f8-40fd-9c4a-56a341c03dcf)

Reply:
It turns out that upgrading the structure of the workflow (namely, adding `workflow-dispatch` capabilities) naturally also solved the `refs/` problem, too.

In other scripts, we use the manually-specified release name as the tag, which has the benefit of resolving conflicts naturally (since releases are typically chosen via the current date).

Reply:
I think this is good to merge? Though, I'll manually run a release after merging just to double-check that everything works as intended.